test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Parsers/ODataPathParserTests.cs

```diff
@@ -36,30 +36,72 @@ public void CallbackReturnsBatchReferenceSegment()
 
         [Fact]
         public void RequestUriProcessorExtractSegmentIdentifierTest()
-        {
-            ExtractSegmentIdentifierAndParenthesisExpression("blah", "blah", null);
-            ExtractSegmentIdentifierAndParenthesisExpression("multiple words", "multiple words", null);
-            ExtractSegmentIdentifierAndParenthesisExpression("multiple(123)", "multiple", "123");
-            ExtractSegmentIdentifierAndParenthesisExpression("multiple(123;321)", "multiple", "123;321");
-            ExtractSegmentIdentifierAndParenthesisExpression("set()", "set", string.Empty);
-        }
-
-        [Fact]
-        public void RequestUriProcessorExtractSegmentIdentifierErrorTest()
         {
             string actualIdentifier;
             string queryPortion;
 
-            Action emptyString = () => ODataPathParser.ExtractSegmentIdentifierAndParenthesisExpression(string.Empty, out actualIdentifier, out queryPortion);
+            // Success cases
+            ExtractSegmentIdentifierAndParenthesisExpression(enableKeyAsSegment: false, "blah", "blah", null);
+            ExtractSegmentIdentifierAndParenthesisExpression(enableKeyAsSegment: false, "multiple words", "multiple words", null);
+            ExtractSegmentIdentifierAndParenthesisExpression(enableKeyAsSegment: false, "multiple(123)", "multiple", "123");
+            ExtractSegmentIdentifierAndParenthesisExpression(enableKeyAsSegment: false, "multiple(123;321)", "multiple", "123;321");
+            ExtractSegmentIdentifierAndParenthesisExpression(enableKeyAsSegment: false, "set()", "set", string.Empty);

```

Comment:
Could you add a test case for a scenario that does not contain an `(` but ends with a `)`, like `blah)`, I expect the result would be `blah)` for the identifier and `null` for the parenthesis expression.